source/code/plugins/oms_configuration.rb

```diff
@@ -167,9 +213,20 @@ def update_azure_resource_id()
             break if retries > max_retries
             azure_resource_id = get_azure_resid_from_imds()
```

Comment:
Might be cleaner if your first action after the `break if` is to check whether or not you are in the hybrid env by checking `azcmagent show`, and then based on that you can make the call to either `get_azure_resid_from_imds` or `get_azure_resid_from_hybridagent`

That way you can avoid the sleep/increment retries code.

Reply:
Sure. Good idea. I'll add that change and update the PR